Geometry2d/Point.hpp

```diff
@@ -157,6 +163,24 @@ class Point {
         return x() != other.x() || y() != other.y();
     }
 
+    double operator[](int i) {
+        if (0 == i) {
+            return _x;
+        } else if (1 == i) {
+            return _y;
+        } else {
+            throw std::out_of_range("Out of range index for Geometry2d::Point");
+        }
+    }
+
+    /**
+     * Hash function for Geometry2d::Point
+     */
+    static size_t hash(Point pt) {
+        return std::hash<std::string>{}(std::to_string(pt.x()) + " " + std::to_string(pt.y()));
```

Comment:
Converting to string is very very slow and sort of makes the entire hash idea pointless. Maybe just call std::hash on the sum of x and y or x multiplied by some big prime plus y?